proton/api.py

```diff
@@ -87,11 +100,23 @@ def api_request(self, endpoint, jsondata=None, additional_headers=None, method=N
         if fct is None:
             raise ValueError("Unknown method: {}".format(method))
 
-        ret = fct(
-            self.__api_url + endpoint,
-            headers = additional_headers,
-            json = jsondata
-        ).json()
+        try:
+            ret = fct(
+                self.__api_url + endpoint,
+                headers = additional_headers,
+                json = jsondata
+            )
+            ret.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            raise ProtonError(
+                {
+                    "Code": e.response.status_code,
+                    "Error": e.response.reason,
+                    "Header": e.response.headers
+                }
+            )
```

Comment:
Wait. Here you would raise HTTP error and discard JSON response for any 4XX or 5XX errors.